packages/react-components/react-card/src/stories/Card/CardInteractive.stories.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { makeStyles, shorthands } from '@fluentui/react-components';
+
+import { Title } from './SampleCard.stories';
+import { SampleCardSmall } from './SampleCardSmall.stories';
+
+const useStyles = makeStyles({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    ...shorthands.gap('30px'),
+  },
+});
+
+export const Interactive = () => {
+  const styles = useStyles();
+
+  const onClickEvent = React.useCallback(() => {
+    console.log('Interactive when has onClick event');
+  }, []);
+
+  return (
+    <div className={styles.root}>
+      <div>
+        <Title title="As a link" />
+        <SampleCardSmall as="a" href="#" />
+      </div>
+
+      <div>
+        <Title title="As a button" />
+        <SampleCardSmall as="button" type="button" />
+      </div>
+
+      <div>
+        <Title title="With a MouseEvent" />
+        <SampleCardSmall onClick={onClickEvent} />
```

Comment:
I believe we could remove this option and solely depend on either `role=button` or `as=button | a` for this!, I mean, if it's interactive, it should be a `button`, otherwise this might be confusing on the a11y POV